tests/v2/validation/auth/openldap.go

```diff
@@ -0,0 +1,43 @@
+package auth
+
+import (
+	"context"
+	"time"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	v3 "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	passwordSecretID       = "cattle-global-data/openldapconfig-serviceaccountpassword"
+	authProviderAnnotation = "management.cattle.io/auth-provider-cleanup"
+	locked                 = "rancher-locked"
+	unlocked               = "unlocked"
+)
+
+func waitUntilAnnotationIsUpdated(client *rancher.Client) (*v3.AuthConfig, error) {
+	ldapConfig, err := client.Management.AuthConfig.ByID("openldap")
+	if err != nil {
+		return nil, err
+	}
+
+	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, 2*time.Minute, true, func(context.Context) (bool, error) {
+		newLDAPConfig, err := client.Management.AuthConfig.ByID("openldap")
+		if err != nil {
+			return false, nil
+		}
+
+		if ldapConfig.Annotations[authProviderAnnotation] != newLDAPConfig.Annotations[authProviderAnnotation] {
```

Comment:
If the auth provider annotations are updated before reaching the PollUntilContextTimeout function, there's a risk that the condition being polled might always be false, that could lead to time out after 2 minutes. isn't it?

Reply:
So this poll was necessary cause this annotation update doesn't get into place like enabled true or false. Seems like, the annotation of the object depends on other handlers. This poll makes sure the previous annotation and current annotation aren't the same, and yes it'll wait only for 2 mins at makes. But it doesn't assert anything here, it happens on the test side.

Reply:
Looks like cleanup controller is triggered and then it cleans up a bunch of stuff that auth provider creates. The annotation is changed only when the cleanup finishes. 

> there is a cleanup controller that fires on disable that cleans up a bunch of things that probably takes a few seconds.